builder/tencentcloud/cvm/access_config.go

```diff
@@ -50,28 +70,67 @@ var ValidRegions = []Region{
 	Frankfurt, Ashburn, Siliconvalley, Toronto, SaoPaulo,
 }
 
+var packerConfig map[string]interface{}
```

Comment:
This confused me a little. As it is being referenced within common but being initialized in `getConfigFromProfile`. Is there a case where the code after line [271](https://github.com/hashicorp/packer-plugin-tencentcloud/pull/114/files#diff-65a27a9ac5994b632d62a75173859143073c86047530d20c6f7bae14b4724088R271) is called but the map is still empty? 

Since you are reusing this variable for the stored profile data should there be a separate method that is called at least once that can be used by  `getConfigFromProfile` and `packerConfigClient`? 

```go
func loadConfigProfile(cf *TencentCloudAccessConfig) (map[string]interface{}, error) {
if len(packerConfig) > 0 {
   return packerConfig, nil
} 

var (
		profile              string
		sharedCredentialsDir string
		credentialPath       string
		configurePath        string
	)

	if cf.Profile != "" {
		profile = cf.Profile
	} else {
		profile = DEFAULT_PROFILE
	}

	if cf.SharedCredentialsDir != "" {
		sharedCredentialsDir = cf.SharedCredentialsDir
	}

	tmpSharedCredentialsDir, err := homedir.Expand(sharedCredentialsDir)
	if err != nil {
		return nil, err
	}

	if tmpSharedCredentialsDir == "" {
		credentialPath = fmt.Sprintf("%s/.tccli/%s.credential", os.Getenv("HOME"), profile)
		configurePath = fmt.Sprintf("%s/.tccli/%s.configure", os.Getenv("HOME"), profile)
		if runtime.GOOS == "windows" {
			credentialPath = fmt.Sprintf("%s/.tccli/%s.credential", os.Getenv("USERPROFILE"), profile)
			configurePath = fmt.Sprintf("%s/.tccli/%s.configure", os.Getenv("USERPROFILE"), profile)
		}
	} else {
		credentialPath = fmt.Sprintf("%s/%s.credential", tmpSharedCredentialsDir, profile)
		configurePath = fmt.Sprintf("%s/%s.configure", tmpSharedCredentialsDir, profile)
	}

	packerConfig = make(map[string]interface{})
	_, err = os.Stat(credentialPath)
	if !os.IsNotExist(err) {
		data, err := ioutil.ReadFile(credentialPath)
		if err != nil {
			return nil, err
		}

		config := map[string]interface{}{}
		err = json.Unmarshal(data, &config)
		if err != nil {
			return nil, err
		}

		for k, v := range config {
			packerConfig[k] = strings.TrimSpace(v.(string))
		}
	} else {
		return nil, err
	}
	_, err = os.Stat(configurePath)
	if !os.IsNotExist(err) {
		data, err := ioutil.ReadFile(configurePath)
		if err != nil {
			return nil, err
		}

		config := map[string]interface{}{}
		err = json.Unmarshal(data, &config)
		if err != nil {
			return nil, err
		}

	outerLoop:
		for k, v := range config {
			if k == "_sys_param" {
				tmpMap := v.(map[string]interface{})
				for tmpK, tmpV := range tmpMap {
					if tmpK == "region" {
						packerConfig[tmpK] = strings.TrimSpace(tmpV.(string))
						break outerLoop
					}
				}
			}
		}
	} else {
		return nil, err
	}

	return packerConfig, nil
}
```

Then the following function code be called in getConfigFromProfile

```go
func (cf *TencentCloudAccessConfig) GetProfileConfig( name string) (interface{}, error) {
   cfg, err := loadConfigProfile(cf)
    if err != nil {
      return nil, err
   }
   profile, ok := cfg[name]
   if !ok {
     return nil, fmt.Errorf( "...")
   }
   return profile, nil
}

Then inside of the packerConfigClient method you could do something like 

func packerConfigClient(cf *TencentCloudAccessConfig) (*TencentCloudClient, error) 
//redacted 
cfg, err := loadConfigProfile(cf)
  if err != nil {
   return nil, err
}
if cf.AssumeRole.RoleArn == "" && cfg["role-arn"] != nil {
		cf.AssumeRole.RoleArn = cfg["role-arn"].(string)
	}

	if cf.AssumeRole.SessionName == "" && cfg["role-session-name"] != nil {
		cf.AssumeRole.SessionName = cfg["role-session-name"].(string)
	}
...
}